crates/nu-command/src/strings/base/base64.rs

```diff
@@ -0,0 +1,193 @@
+use data_encoding::Encoding;
+
+use nu_engine::command_prelude::*;
+
+const EXTRA_USAGE: &str = r"The default alphabet is taken from RFC 4648, section 4.  A URL-safe version is available.
+
+Note this command will collect stream input.";
+
+fn get_encoding_from_flags(url: bool, nopad: bool) -> Encoding {
+    match (url, nopad) {
+        (false, false) => data_encoding::BASE64,
+        (false, true) => data_encoding::BASE64_NOPAD,
+        (true, false) => data_encoding::BASE64URL,
+        (true, true) => data_encoding::BASE64URL_NOPAD,
+    }
+}
+
+fn get_encoding(
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+) -> Result<Encoding, ShellError> {
+    let url = call.has_flag(engine_state, stack, "url")?;
+    let nopad = call.has_flag(engine_state, stack, "nopad")?;
+
+    Ok(get_encoding_from_flags(url, nopad))
+}
+
+fn get_encoding_const(working_set: &StateWorkingSet, call: &Call) -> Result<Encoding, ShellError> {
+    let url = call.has_flag_const(working_set, "url")?;
+    let nopad = call.has_flag_const(working_set, "nopad")?;
+
+    Ok(get_encoding_from_flags(url, nopad))
+}
+
+#[derive(Clone)]
+pub struct DecodeBase64;
+
+impl Command for DecodeBase64 {
+    fn name(&self) -> &str {
+        "decode new-base64"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("decode new-base64")
+            .input_output_types(vec![(Type::String, Type::Binary)])
+            .allow_variants_without_examples(true)
+            .switch("url", "Decode the URL-safe Base64 version.", None)
+            .switch("nopad", "Reject padding.", None)
+            .category(Category::Formats)
+    }
+
+    fn usage(&self) -> &str {
+        "Decode a Base64 value."
+    }
+
+    fn extra_usage(&self) -> &str {
+        EXTRA_USAGE
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Decode a Base64 string",
+                example: r#""U29tZSBEYXRh" | decode new-base64 | decode"#,
+                result: None,
+            },
+            Example {
+                description: "Decode arbitrary data",
+                example: r#""/w==" | decode new-base64"#,
+                result: Some(Value::test_binary(vec![0xFF])),
+            },
+            Example {
+                description: "Decode a URL-safe Base64 string",
```

Comment:
The output doesn't appear to be a string.

Reply:
Yeah, by default decoding returns a binary, because encoding might not be a string (base64 images in a JSON API is the first example which comes to mind).  So, the signature is `string -> binary`.  That's why the `new-` thing is needed, because it's incompatible with the old `decode base64`[^hex].

[^hex]: and the `decode hex` command was already decoding to binary, so I could keep compatibility there.